internal/handlers/common/update.go

```diff
@@ -50,6 +50,20 @@ func UpdateDocument(doc, update *types.Document) (bool, error) {
 			}
 			changed = true
 
+		case "$unset":
+			switch unsetDoc := updateV.(type) {
+			case *types.Document:
+				if unsetDoc.Len() == 0 {
+					continue
+				}
+				for _, key := range unsetDoc.Keys() {
+					doc.Remove(key)
+				}
+				changed = true
+			default:
+				panic("Modifiers operate on fields but we found another type instead")
```

Comment:
Why ’panic’? Maybe just return error?

Reply:
It that case switch could be replaced with type assertion.

Reply:
> It that case switch could be replaced with type assertion.

Good to note! Changed here and will change in the near PR. Thank you, @w84thesun!